### DISCUSSION TITLE
Allow using files/images from HTTP request/webscraper/other nodes as the context/input of an LLM node that support vision

### DISCUSSION THREAD

**Discussion by pjamessteven** [langgenius/dify#4323] [Suggestion]
This would be a really powerful feature for workflow mode. It feels quite restrictive the way dify handles files and images at the moment. You can't do much with them. It looks like a lot of the groundwork is already in the front end code base at least. If I find the time I might try work on this but otherwise just putting this idea out there.

One thing I would love to do with dify is use a webscraper or web-screengrab component to get an image of a full webpage, and then pass it to an LLM node with vision to ask it questions about the page. I find that chat-gpt's vision capabilities are much more capable than the current webscraper tool (as long as you tell it not to fall back to the python OCR package).

**Comment by Lagyu** [comment]
+1
I really want to fully make use of gpt-4o or other models with image inputs from external resources.

**Comment by MatheMatrix** [comment]
+1
